client/src/message_interface/message_handler.rs

```diff
@@ -38,17 +38,32 @@ use crate::{
 };
 
 fn panic_to_response_message(panic: Box<dyn Any>) -> Response {
-    let msg = if let Some(message) = panic.downcast_ref::<String>() {
-        format!("Internal error: {message}")
-    } else if let Some(message) = panic.downcast_ref::<&str>() {
-        format!("Internal error: {message}")
-    } else {
-        "Internal error".to_string()
-    };
+    let msg = panic.downcast_ref::<String>().map_or_else(
+        || {
+            panic.downcast_ref::<&str>().map_or_else(
+                || "Internal error".to_string(),
+                |message| format!("Internal error: {message}"),
+            )
+        },
+        |message| format!("Internal error: {message}"),
+    );
     let current_backtrace = Backtrace::new();
     Response::Panic(format!("{msg}\n\n{current_backtrace:?}"))
 }
 
+#[cfg(not(target_family = "wasm"))]
+async fn convert_async_panics<F>(f: impl FnOnce() -> F + Send) -> Result<Response>
+where
+    F: Future<Output = Result<Response>> + Send,
+{
+    match AssertUnwindSafe(f()).catch_unwind().await {
+        Ok(result) => result,
+        Err(panic) => Ok(panic_to_response_message(panic)),
+    }
```

Comment:
Is the following better?
```suggestion
    AssertUnwindSafe(f())
        .catch_unwind()
        .await
        .unwrap_or_else(|panic| Ok(panic_to_response_message(panic)))
```

Reply:
Changed